app/controllers/Authentication.scala

```diff
@@ -358,6 +365,78 @@ class Authentication @Inject()(
+    signUpForm.bindFromRequest.fold(
+      bogusForm => Future.successful(BadRequest(bogusForm.toString)),
+      signUpData => {
+        creatingOrganizationsIsAllowed(request.identity).futureBox.flatMap {
+          case Full(_) =>
+            val email = signUpData.email.toLowerCase
+            val loginInfo = LoginInfo(CredentialsProvider.ID, email)
+            var errors = List[String]()
+            val firstName = normalizeName(signUpData.firstName).getOrElse {
+              errors ::= Messages("user.firstName.invalid"); ""
+            }
+            val lastName = normalizeName(signUpData.lastName).getOrElse {
+              errors ::= Messages("user.lastName.invalid"); ""
+            }
+            UserService.retrieve(loginInfo).toFox.futureBox.flatMap {
+              case Full(_) =>
+                errors ::= Messages("user.email.alreadyInUse")
+                Fox.successful(BadRequest(Json.obj("messages" -> Json.toJson(errors.map(t => Json.obj("error" -> t))))))
+              case Empty =>
+                if (errors.nonEmpty) {
+                  Fox.successful(BadRequest(Json.obj("messages" -> Json.toJson(errors.map(t => Json.obj("error" -> t))))))
+                } else {
+                  for {
+                    organization <- createOrganization(signUpData.organization) ?~> Messages("organization.create.failed")
+                    user <- UserService.insert(organization.name, email, firstName, lastName, signUpData.password, isActive = true, teamRole = true,
+                      loginInfo, passwordHasher.hash(signUpData.password), isAdmin = true)
+                    _ <- createOrganizationFolder(signUpData.organization, loginInfo)
+                  } yield Ok
+                }
+              case f: Failure => Fox.failure(f.msg)
+            }
+          case _ => Fox.failure(Messages("organization.create.forbidden"))
+        }
+      }
+    )
+  }
+
+  private def creatingOrganizationsIsAllowed(requestingUser: Option[User]) = {
+    val noOrganizationPresent = InitialDataService.assertNoOrganizationsPresent
+    val configurationFlagSet = Play.configuration.getBoolean("application.allowOrganzationCreation").getOrElse(false) ?~> "allowOrganzationCreation.notEnabled"
+    val userIsSuperUser = requestingUser.exists(_.isSuperUser).toFox
+
+    Fox.sequenceOfFulls(List(noOrganizationPresent, configurationFlagSet, userIsSuperUser)).map(_.headOption).toFox
+  }
+
+  private def createOrganization(organizationName: String) = {
+    val organization = OrganizationSQL(ObjectId.fromBsonId(BSONObjectID.generate), organizationName, "", "")
+    val organizationTeam = TeamSQL(ObjectId.fromBsonId(BSONObjectID.generate), organization._id, organization.name, isOrganizationTeam = true)
+    for {
+      _ <- OrganizationSQLDAO.insertOne(organization)(GlobalAccessContext)
+      _ <- TeamSQLDAO.insertOne(organizationTeam)(GlobalAccessContext)
+      _ <- InitialDataService.insertLocalDataStoreIfEnabled
```

Comment:
Why insert a local datastore to the db here? I guess I might be missing a case. @youri-k